test/test_unistd.rs

```diff
@@ -29,23 +29,26 @@ fn test_fork_and_waitpid() {
                 Ok(WaitStatus::Exited(pid_t, _)) =>  assert!(pid_t == child),
 
                 // panic, must never happen
-                Ok(_) => panic!("Child still alive, should never happen"),
+                s @ Ok(_) => panic!("Child exited {:?}, should never happen", s),
```

Comment:
I don't think so.  The examples for `unreachable!` are for stuff that's so obviously unreachable that a smarter compiler should be able to figure it out.  But in this case, the assertion can actually be hit; for example if you SIGKILL the child.  I actually did hit this assertion frequently when my child deadlocked and had to be killed.